BUG: osx-64 scipy 1.9.1 test_bad_geneig numerical error

### Describe your issue.

The test test_bad_geneig is failing on scipy 1.9.1 osx-64 with openblas 0.3.21 for py38, py39 and py310.
It looks like a permutation issue.

### Reproducing Code Example

```python
This is reproducible on osx-64 with openblas 0.3.21 for py38, py39 and py310.

python -c 'import scipy, sys; scipy.test(verbose=2, label='\''full'\'', tests=None, extra_argv=['\''-k'\'', '\''(test_bad_geneig)'\'', '\''-n'\'', '\''3'\'', '\''--timeout=1800'\'', '\''--durations=50'\'']); sys.exit(0)'
```


### Error message

```shell
INFO  [gw0] darwin -- Python 3.10.4 $PREFIX/bin/python                                                                                                                                           
         INFO  $PREFIX/lib/python3.10/site-packages/scipy/linalg/tests/test_decomp.py:337: in test_bad_geneig                                                                                             
         INFO      self._check_gen_eig(A, B)                                                                                                                                                              
         INFO          A          = [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, -8.9375, 0], [0, 0, 0, -8.9375]]                                                                                                  
         INFO          B          = [[0, 0, 1, 0], [0, 0, 0, 1], [1, 0, 0, -0.5], [0, 1, 0.5, 0]]                                                                                                         
         INFO          k          = 5                                                                                                                                                                     
         INFO          matrices   = <function TestEig.test_bad_geneig.<locals>.matrices at 0x7fc4988a2560>                                                                                                
         INFO          self       = <scipy.linalg.tests.test_decomp.TestEig object at 0x7fc49970d510>                                                                                                     
         INFO  $PREFIX/lib/python3.10/site-packages/scipy/linalg/tests/test_decomp.py:274: in _check_gen_eig                                                                                              
         INFO      assert_allclose(w[perm], wt[permt],                                                                                                                                                    
         INFO  E   AssertionError:                                                                                                                                                                        
         INFO  E   Not equal to tolerance rtol=1e-07, atol=1e-07                                                                                                                                          
         INFO  E                                                                                                                                                                                          
         INFO  E   array([[ 1.    ,  0.    ,  0.    ,  0.    ],                                                                                                                                           
         INFO  E          [ 0.    ,  1.    ,  0.    ,  0.    ],                                                                                                                                           
         INFO  E          [ 0.    ,  0.    , -8.9375,  0.    ],                                                                                                                                           
         INFO  E          [ 0.    ,  0.    ,  0.    , -8.9375]])                                                                                                                                          
         INFO  E   array([[ 0. ,  0. ,  1. ,  0. ],                                                                                                                                                       
         INFO  E          [ 0. ,  0. ,  0. ,  1. ],                                                                                                                                                       
         INFO  E          [ 1. ,  0. ,  0. , -0.5],                                                                                                                                                       
         INFO  E          [ 0. ,  1. ,  0.5,  0. ]])                                                                                                                                                      
         INFO  E   Mismatched elements: 3 / 4 (75%)                                                                                                                                                       
         INFO  E   Max absolute difference: 6.                                                                                                                                                            
         INFO  E   Max relative difference: 2.18181818                                                                                                                                                    
         INFO  E    x: array([-3.866788e-17-2.75j, -3.866788e-17+2.75j,  0.000000e+00-3.25j,                                                                                                              
         INFO  E           0.000000e+00+3.25j])                                                                                                                                                           
         INFO  E    y: array([0.-3.25j, 0.-2.75j, 0.+2.75j, 0.+3.25j])                                                                                                                                    
         INFO          A          = array([[ 1.    ,  0.    ,  0.    ,  0.    ],                                                                                                                          
         INFO         [ 0.    ,  1.    ,  0.    ,  0.    ],                                                                                                                                               
         INFO         [ 0.    ,  0.    , -8.9375,  0.    ],                                                                                                                                               
         INFO         [ 0.    ,  0.    ,  0.    , -8.9375]])                                                                                                                                              
         INFO          B          = array([[ 0. ,  0. ,  1. ,  0. ],                                                                                                                                      
         INFO         [ 0. ,  0. ,  0. ,  1. ],                                                                                                                                                           
         INFO         [ 1. ,  0. ,  0. , -0.5],                                                                                                                                                           
         INFO         [ 0. ,  1. ,  0.5,  0. ]])                                                                                                                                                          
         INFO          B0         = array([[ 0. ,  0. ,  1. ,  0. ],                                                                                                                                      
         INFO         [ 0. ,  0. ,  0. ,  1. ],                                                                                                                                                           
         INFO         [ 1. ,  0. ,  0. , -0.5],                                                                                                                                                           
         INFO         [ 0. ,  1. ,  0.5,  0. ]])                                                                                                                                                          
         INFO          beta_nonzero = array([ True,  True,  True,  True])                                                                                                                                 
         INFO          i          = 3                                                                                                                                                                     
         INFO          length     = array([1., 1., 1., 1.])                                                                                                                                               
         INFO          msg        = '\narray([[ 1.    ,  0.    ,  0.    ,  0.    ],\n       [ 0.    ,  1.    ,  0.    ,  0.    ],\n       [ 0.    ,  0.   ... ,  1. ,  0. ],\n       [ 0. ,  0. ,  0. ,   
               1. ],\n       [ 1. ,  0. ,  0. , -0.5],\n       [ 0. ,  1. ,  0.5,  0. ]])'                                                                                                                
         INFO          perm       = array([3, 2, 1, 0])                                                                                                                                                   
         INFO          permt      = array([1, 3, 2, 0])                                                                                                                                                   
         INFO          res        = array([[ 5.55111512e-16+0.00000000e+00j,  5.55111512e-16+0.00000000e+00j,                                                                                             
         INFO           7.98297505e-17-4.44089210e-16j,  7....0000e+00j,  8.88178420e-16+0.00000000e+00j,                                                                                                 
         INFO          -3.89981295e-17+0.00000000e+00j, -3.89981295e-17+0.00000000e+00j]])                                                                                                                
         INFO          self       = <scipy.linalg.tests.test_decomp.TestEig object at 0x7fc49970d510>                                                                                                     
         INFO          val1       = array([[-6.75837818e-01+0.00000000e+00j, -6.75837818e-01+0.00000000e+00j,                                                                                             
         INFO           9.65826743e-17+6.64534303e-01j,  9....0000e+00j,  1.85855400e+00+0.00000000e+00j,                                                                                                 
         INFO          -5.64792532e-16+2.15973648e+00j, -5.64792532e-16-2.15973648e+00j]])                                                                                                                
         INFO          val2       = array([[-6.75837818e-01+0.00000000e+00j, -6.75837818e-01-0.00000000e+00j,                                                                                             
         INFO           1.67529238e-17+6.64534303e-01j,  1....0000e+00j,  1.85855400e+00+0.00000000e+00j,                                                                                                 
         INFO          -5.25794402e-16+2.15973648e+00j, -5.25794402e-16-2.15973648e+00j]])                                                                                                                
         INFO          vr         = array([[-6.75837818e-01+0.00000000e+00j, -6.75837818e-01+0.00000000e+00j,                                                                                             
         INFO           9.65826743e-17+6.64534303e-01j,  9....0000e+00j, -2.07950098e-01+0.00000000e+00j,                                                                                                 
         INFO           6.31935700e-17-2.41648837e-01j,  6.31935700e-17+2.41648837e-01j]])                                                                                                                
         INFO          w          = array([ 0.00000000e+00+3.25j,  0.00000000e+00-3.25j,                                                                                                                  
         INFO         -3.86678798e-17+2.75j, -3.86678798e-17-2.75j])                                                                                                                                      
         INFO          w_fin      = array([ 0.00000000e+00+3.25j,  0.00000000e+00-3.25j,                                                                                                                  
         INFO         -3.86678798e-17+2.75j, -3.86678798e-17-2.75j])                                                                                                                                      
         INFO          wh         = array([ 0.00000000e+00+3.25j,  0.00000000e+00-3.25j,                                                                                                                  
         INFO         -3.86678798e-17+2.75j, -3.86678798e-17-2.75j])                                                                                                                                      
         INFO          wt         = array([0.+3.25j, 0.-3.25j, 0.+2.75j, 0.-2.75j])                                                                                                                       
         INFO          wt_fin     = array([0.+3.25j, 0.-3.25j, 0.+2.75j, 0.-2.75j])
```


### SciPy/NumPy/Python version information

1.9.1 1.23.1 sys.version_info(major=3, minor=10, micro=4, releaselevel='final', serial=0)

This is still an issue as of 1.11.0 in [conda-forge](https://github.com/conda-forge/scipy-feedstock/pull/242), though only for windows + newest openblas (0.3.23).

<details>
<summary>Stacktrace</summary>

```
================================== FAILURES ===================================
___________________________ TestEig.test_bad_geneig ___________________________
[gw1] win32 -- Python 3.10.12 %PREFIX%\python.exe
..\_test_env\lib\site-packages\scipy\linalg\tests\test_decomp.py:310: in test_bad_geneig
    self._check_gen_eig(A, B)
        A          = [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, -8.9975, 0], [0, 0, 0, -8.9975]]
        B          = [[0, 0, 1, 0], [0, 0, 0, 1], [1, 0, 0, -0.1], [0, 1, 0.1, 0]]
        k          = 1
        matrices   = <function TestEig.test_bad_geneig.<locals>.matrices at 0x000002BD8AE96950>
        self       = <scipy.linalg.tests.test_decomp.TestEig object at 0x000002BDF3109B40>
..\_test_env\lib\site-packages\scipy\linalg\tests\test_decomp.py:247: in _check_gen_eig
    assert_allclose(w[perm], wt[permt],
        A          = array([[ 1.    ,  0.    ,  0.    ,  0.    ],
       [ 0.    ,  1.    ,  0.    ,  0.    ],
       [ 0.    ,  0.    , -8.9975,  0.    ],
       [ 0.    ,  0.    ,  0.    , -8.9975]])
        B          = array([[ 0. ,  0. ,  1. ,  0. ],
       [ 0. ,  0. ,  0. ,  1. ],
       [ 1. ,  0. ,  0. , -0.1],
       [ 0. ,  1. ,  0.1,  0. ]])
        B0         = array([[ 0. ,  0. ,  1. ,  0. ],
       [ 0. ,  0. ,  0. ,  1. ],
       [ 1. ,  0. ,  0. , -0.1],
       [ 0. ,  1. ,  0.1,  0. ]])
        beta_nonzero = array([ True,  True,  True,  True])
        i          = 3
        length     = array([1., 1., 1., 1.])
        msg        = '\narray([[ 1.    ,  0.    ,  0.    ,  0.    ],\n       [ 0.    ,  1.    ,  0.    ,  0.    ],\n       [ 0.    ,  0.   ... ,  1. ,  0. ],\n       [ 0. ,  0. ,  0. ,  1. ],\n       [ 1. ,  0. ,  0. , -0.1],\n       [ 0. ,  1. ,  0.1,  0. ]])'
        perm       = array([1, 0, 3, 2], dtype=int64)
        permt      = array([1, 3, 2, 0], dtype=int64)
        res        = array([[ 0.00000000e+00-1.11022302e-16j,  0.00000000e+00+1.11022302e-16j,
        -7.77156117e-16-3.96947546e-17j, -7....9065e-15j,  0.00000000e+00-2.44249065e-15j,
         4.44089210e-16+6.52076712e-17j,  4.44089210e-16-6.52076712e-17j]])
        self       = <scipy.linalg.tests.test_decomp.TestEig object at 0x000002BDF3109B40>
        val1       = array([[ 0.00000000e+00-6.71913752e-01j,  0.00000000e+00+6.71913752e-01j,
        -6.69676706e-01+1.21957074e-16j, -6....4557e+00j,  0.00000000e+00-1.98214557e+00j,
        -2.04251395e+00+3.44054481e-16j, -2.04251395e+00-3.44054481e-16j]])
        val2       = array([[-0.00000000e+00-6.71913752e-01j,  0.00000000e+00+6.71913752e-01j,
        -6.69676706e-01+1.61651829e-16j, -6....4557e+00j,  0.00000000e+00-1.98214557e+00j,
        -2.04251395e+00+2.78846810e-16j, -2.04251395e+00-2.78846810e-16j]])
        vr         = array([[ 0.00000000e+00-6.71913752e-01j,  0.00000000e+00+6.71913752e-01j,
        -6.69676706e-01+1.21957074e-16j, -6....9591e-01j,  0.00000000e+00+2.20299591e-01j,
         2.27009053e-01-3.82388976e-17j,  2.27009053e-01+3.82388976e-17j]])
        w          = array([0.00000000e+00+3.05j, 0.00000000e+00-3.05j, 1.00194251e-16+2.95j,
       1.00194251e-16-2.95j])
        w_fin      = array([0.00000000e+00+3.05j, 0.00000000e+00-3.05j, 1.00194251e-16+2.95j,
       1.00194251e-16-2.95j])
        wh         = array([0.00000000e+00+3.05j, 0.00000000e+00-3.05j, 1.00194251e-16+2.95j,
       1.00194251e-16-2.95j])
        wt         = array([0.+3.05j, 0.-3.05j, 0.+2.95j, 0.-2.95j])
        wt_fin     = array([0.+3.05j, 0.-3.05j, 0.+2.95j, 0.-2.95j])
..\_test_env\lib\contextlib.py:79: in inner
    return func(*args, **kwds)
E   AssertionError:
E   Not equal to tolerance rtol=1e-07, atol=1e-07
E
E   array([[ 1.    ,  0.    ,  0.    ,  0.    ],
E          [ 0.    ,  1.    ,  0.    ,  0.    ],
E          [ 0.    ,  0.    , -8.9975,  0.    ],
E          [ 0.    ,  0.    ,  0.    , -8.9975]])
E   array([[ 0. ,  0. ,  1. ,  0. ],
E          [ 0. ,  0. ,  0. ,  1. ],
E          [ 1. ,  0. ,  0. , -0.1],
E          [ 0. ,  1. ,  0.1,  0. ]])
E   Mismatched elements: 3 / 4 (75%)
E   Max absolute difference: 6.
E   Max relative difference: 2.03389831
E    x: array([0.000000e+00-3.05j, 0.000000e+00+3.05j, 1.001943e-16-2.95j,
E          1.001943e-16+2.95j])
E    y: array([0.-3.05j, 0.-2.95j, 0.+2.95j, 0.+3.05j])
        args       = (<function assert_allclose.<locals>.compare at 0x000002BD8AE95FC0>, array([0.00000000e+00-3.05j, 0.00000000e+00+3.05j, 1.00194251e-16-2.95j,
       1.00194251e-16+2.95j]), array([0.-3.05j, 0.-2.95j, 0.+2.95j, 0.+3.05j]))
        func       = <function assert_array_compare at 0x000002BDDD30F010>
        kwds       = {'equal_nan': True, 'err_msg': '\narray([[ 1.    ,  0.    ,  0.    ,  0.    ],\n       [ 0.    ,  1.    ,  0.    ,  0....-0.1],\n       [ 0. ,  1. ,  0.1,  0. ]])', 'header': 'Not equal to tolerance rtol=1e-07, atol=1e-07', 'verbose': True}
        self       = <contextlib._GeneratorContextManager object at 0x000002BDDD324CA0>
```

</details>

--- a/scipy/linalg/tests/test_decomp.py
+++ b/scipy/linalg/tests/test_decomp.py
@@ -59,22 +59,6 @@ COMPLEX_DTYPES = [np.complex64, np.complex128]
 DTYPES = REAL_DTYPES + COMPLEX_DTYPES
 
 
-def clear_fuss(ar, fuss_binary_bits=7):
-    """Clears trailing `fuss_binary_bits` of mantissa of a floating number"""
-    x = np.asanyarray(ar)
-    if np.iscomplexobj(x):
-        return clear_fuss(x.real) + 1j * clear_fuss(x.imag)
-
-    significant_binary_bits = np.finfo(x.dtype).nmant
-    x_mant, x_exp = np.frexp(x)
-    f = 2.0**(significant_binary_bits - fuss_binary_bits)
-    x_mant *= f
-    np.rint(x_mant, out=x_mant)
-    x_mant /= f
-
-    return np.ldexp(x_mant, x_exp)
-
-
 # XXX: This function should not be defined here, but somewhere in
 #      scipy.linalg namespace
 def symrand(dim_or_eigv, rng):
@@ -238,11 +222,18 @@ class TestEig:
                 assert_allclose(res[:, i], 0,
                                 rtol=1e-13, atol=1e-13, err_msg=msg)
 
+        # try to consistently order eigenvalues, including complex conjugate pairs
         w_fin = w[isfinite(w)]
         wt_fin = wt[isfinite(wt)]
-        perm = argsort(clear_fuss(w_fin))
-        permt = argsort(clear_fuss(wt_fin))
-        assert_allclose(w[perm], wt[permt],
+
+        # prune noise in the real parts
+        w_fin = -1j * np.real_if_close(1j*w_fin, tol=1e-10)
+        wt_fin = -1j * np.real_if_close(1j*wt_fin, tol=1e-10)
+
+        perm = argsort(w_fin)
+        permt = argsort(wt_fin)
+
+        assert_allclose(w_fin[perm], wt_fin[permt],
                         atol=1e-7, rtol=1e-7, err_msg=msg)
 
         length = np.empty(len(vr))

